src/Servers/Kestrel/Transport.MsQuic/Internal/QuicSession.cs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using static Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic.Internal.QuicListener;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic.Internal
+{
+    // TODO I still need to finish a small refactor here to remove allocations per connection.
+    internal sealed class QuicSession : IDisposable
+    {
+        private bool _disposed = false;
+        private IntPtr _nativeObjPtr;
+        private byte[] _alpn;
```

Comment:
Why save the ALPN in this object? It doesn't seem to be used. MsQuic is already saving it below.